src/middlewared/middlewared/etc_files/master.passwd.mako

```diff
@@ -1,24 +1,9 @@
-% for user in middleware.call_sync('user.query', [], {'order_by': ['-builtin', 'uid']}):
 <%
-if IS_FREEBSD:
-    if user['password_disabled']:
-        passwd = "*"
-    elif user['locked']:
-        passwd = "*LOCKED*"
-    else:
-        passwd = user['unixhash']
-else:
-    passwd = "x"
+    from middlewared.utils import filter_list
 
-if IS_FREEBSD:
-    freebsd_fields = ":0:0:"
-else:
-    freebsd_fields = ""
+    users = filter_list(render_ctx['user.query'], [], {'order_by': ['-builtin', 'uid']})
 %>\
-${user['username']}:${passwd}:${user['uid']}:${user['group']['bsdgrp_gid']}:${freebsd_fields}${user['full_name']}:${user['home']}:${user['shell']}
+% for user in users:
+${user['username']}:x:${user['uid']}:${user['group']['bsdgrp_gid']}:${user['full_name']}:${user['home']}:${user['shell']}
```

Comment:
What if we have an `'\n'` character in `full_name`? (no need to strip anything here, let's just fix the validation please)

Reply:
I didn't realize that was a valid character. Interesting :)

Reply:
```
bsmith:x:1003:1003:bob \n smith:/mnt/dozer/smb-vss/bsmith:/usr/bin/bash
root@truenas[/mnt/dozer/middleware/src/middlewared]# midclt call user.get_user_obj '{"username": "bsmith"}'
{"pw_name": "bsmith", "pw_uid": 1003, "pw_gid": 1003, "pw_gecos": "bob \\n smith", "pw_dir": "/mnt/dozer/smb-vss/bsmith", "pw_shell": "/usr/bin/bash"}
```

```
root@truenas[/mnt/dozer/middleware/src/middlewared]# midclt call user.query '[["username", "=", "bsmith"]]' '{"get": true}' | jq .full_name
"bob \\n smith"
```